src/nupic/research/spatial_pooler.py

```diff
@@ -1468,7 +1468,7 @@ def _inhibitColumnsLocal(self, overlaps, density):
       overlapSlice = overlaps[maskNeighbors]
       numActive = int(0.5 + density * (len(maskNeighbors) + 1))
       numBigger = numpy.count_nonzero(overlapSlice > overlaps[i])
-      if numBigger < numActive:
+      if overlaps[i] > 0 and numBigger < numActive:
```

Comment:
When it's 0, we're never consuming the `maskNeighbors`, etc., so it'd be better to move this check up a little higher. In other words, there will be two if statements, not just one.